### PR TITLE
generic with configmaps

### PR DESCRIPTION
**Pull request by sosiouxme**
use the generic ES and curator from common and apply configuration as a configmap.

### CODE REVIEW

**Comment by richm** [comment]
lgtm

**Comment by richm** [comment]
[test]

**Comment by sosiouxme** [comment]
try that [test] again, https://ci.openshift.redhat.com/jenkins/job/test-origin-aggregated-logging/347/ broke due to the submodule changes not being merged.

**Comment by sosiouxme** [comment]
@ewolinetz PTAL

tests *might* pass now...

File: deployer/scripts/install.sh [hunk 1]
```diff
@@ -209,7 +208,27 @@ function generate_secrets() {
...
+
+function generate_configmaps() {
+    ### ConfigMaps
+    echo "Deleting configmaps"
+    oc delete configmap -l logging-infra=support
+
+    echo "Creating configmaps"
+
+    # generate elasticsearch configmap 
+    oc create configmap logging-elasticsearch \
```
Comment (ewolinetz) [inline deployer/scripts/install.sh]: how will this work when we upgrade to not clobber any updated configmap?

**Comment by ewolinetz** [comment]
LGTM

File: hack/testing/logging.sh [hunk 2]
```diff
@@ -458,6 +461,29 @@ if [ ! -n "$USE_LOGGING_DEPLOYER_SCRIPT" ] ; then
 fi
 
 #upgrade-test
+function removeEsCuratorConfigMaps() {
```
Comment (ewolinetz) [inline hack/testing/logging.sh]: might want to see how I'm thinking of compressing tests in https://github.com/openshift/origin-aggregated-logging/pull/160

**Comment by sosiouxme** [comment]
There seemed to be some fluke with execing in the fluentd pod in https://ci.openshift.redhat.com/jenkins/job/test-origin-aggregated-logging/360/ - added a little something to catch that sooner if it happens again. [merge]

**Comment by sosiouxme** [comment]
From the pod logs it's clear elasticsearch is really broken, but I really don't understand why. It appears not to have its secret contents, but that shouldn't even be possible, i.e. it shouldn't start at all without its secret. Obviously everything works fine when I run the deployer outside testing. To figure out later...

**Comment by sosiouxme** [comment]
[test]

File: deployer/scripts/install.sh (hunk 1 above)
Comment (sosiouxme) [inline deployer/scripts/install.sh]: I think if we have an upgrade that requires a config change, we have to clobber any changes they may have made. I'm not thinking that we'll encourage customers to change their configmaps; that would be more a debug capability. If there's a special case where they really do need to change them for some legitimate function, they'll have to put their changes back in place after the hypothetical upgrade.

**Comment by sosiouxme** [comment]
random ruby build failure on https://ci.openshift.redhat.com/jenkins/job/test-origin-aggregated-logging/380/console ???

[test] again

**Comment by sosiouxme** [comment]
https://ci.openshift.redhat.com/jenkins/job/test-origin-aggregated-logging/381/ has same failure.
re [test]

**Comment by sosiouxme** [comment]
ruby s2i failure should be fixed, [test]

**Comment by sosiouxme** [comment]
trying again to [merge]

**Comment by sosiouxme** [comment]
[test]

**Comment by sosiouxme** [comment]
[test]

**Comment by sosiouxme** [comment]
[test] again with CI including the submodule change

**Comment by sosiouxme** [comment]
um... [test] again, https://ci.openshift.redhat.com/jenkins/job/test-origin-aggregated-logging/411/ went sideways before doing anything

**Comment by richm** [comment]
[test]

**Comment by richm** [comment]
LGTM

**Comment by sosiouxme** [comment]
[test]

**Comment by sosiouxme** [comment]
[test]

**Comment by sosiouxme** [comment]
[merge]

**Pull request merged**